iron-image.html

```diff
@@ -224,70 +250,76 @@
           type: Number,
           value: null
         },
+      },
 
-        _placeholderBackgroundUrl: {
-          type: String,
-          computed: '_computePlaceholderBackgroundUrl(preload,placeholder)',
-          observer: '_placeholderBackgroundUrlChanged'
-        },
-
-        requiresPreload: {
-          type: Boolean,
-          computed: '_computeRequiresPreload(preload,loaded)'
-        },
-
-        canLoad: {
-          type: Boolean,
-          computed: '_computeCanLoad(preventLoad, src)'
-        }
+      hostAttributes: {
+        role: 'img'
       },
 
       observers: [
-        '_transformChanged(sizing, position)',
-        '_loadBehaviorChanged(canLoad, preload, loaded)',
-        '_loadStateChanged(src, preload, loaded)',
+        '_transformChanged(sizing, position)'
       ],
 
       ready: function() {
-        if (!this.hasAttribute('role')) {
-          this.setAttribute('role', 'img');
-        }
-      },
+        var img = this.$.img;
 
-      _computeImageVisibility: function() {
-        return !!this.sizing;
-      },
+        img.onload = function() {
+          var resolvedSrc = Polymer.ResolveUrl.resolveUrl(this.src, this.ownerDocument.baseURI);
+          if (img.src !== resolvedSrc) return;
```

Comment:
It might be nice to refactor this into a protected method that lets you compare srcs. Something like..

```javascript
if (this._matchesSrc(img.src)) return;
```

WDYT?